mmv1/third_party/terraform/services/resourcemanager/data_source_google_service_account_id_token.go

```diff
@@ -77,12 +77,13 @@ func dataSourceGoogleServiceAccountIdTokenRead(d *schema.ResourceData, meta inte
 		return fmt.Errorf("error calling getCredentials(): %v", err)
 	}
 
-	// If the source credential is not a service account key, use the API to generate the idToken
-	if creds.JSON == nil {
+	targetServiceAccount := d.Get("target_service_account").(string)
```

Comment:
Thanks for fixing the condition here. 
Could `targetServiceAccount  != ""` replace `creds.JSON == nil` here?

Is there some test to cover this particular logic?

Reply:
Parameter`target_service_account` is for impersonation mode. With `target_service_account` you should call an API including service account email regardless of whether JSON is empty or not.

`creds.JSON` is empty when authenticated with GCE metadata server:
https://github.com/golang/oauth2/blob/2323c81c8dba82e8650ed3a24a1a5667e293af38/google/default.go#L160

I wanted to make tests to cover this but I could't find any existing cases emulate GCE metadata server authentication. Is that possible?

Reply:
I don't think I fully get it. What is the relation between in `impersonation mode` and authenticated with GCE metadata server?

Reply:
There is no relation between `impersonation mode` and authenticated with GCE metadata server.

Instances running on GCE are able to authenticate without service account key JSON.
https://google.aip.dev/auth/4115

- `creds.JSON == nil`
  - maybe `impersonation mode`
  - maybe running on GCE
- `targetServiceAccount  == ""`
  - must `impersonation mode`
 
so the condition `creds.JSON == nil` is imperfect.

Reply:
Thanks for the explanation.